packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts

```diff
@@ -38,40 +39,48 @@ export class HostedPluginDeployerHandler implements PluginDeployerHandler {
      */
     private readonly currentFrontendPluginsMetadata: PluginMetadata[] = [];
 
-    getDeployedFrontendMetadata(): PluginMetadata[] {
-        return this.currentFrontendPluginsMetadata;
+    private readonly backendPluginsMetadataDeferred = new Deferred<PluginMetadata[]>();
+
+    private readonly frontendPluginsMetadataDeferred = new Deferred<PluginMetadata[]>();
+
+    getDeployedFrontendMetadata(): Promise<PluginMetadata[]> {
+        return this.frontendPluginsMetadataDeferred.promise;
     }
 
-    getDeployedBackendMetadata(): PluginMetadata[] {
-        return this.currentBackendPluginsMetadata;
+    getDeployedBackendMetadata(): Promise<PluginMetadata[]> {
+        return this.backendPluginsMetadataDeferred.promise;
```

Comment:
you won't be able to install new extensions dynamically with such change

Reply:
@akosyakov Sorry, I don't understand why, could you explain how it can brake dynamic extension installation?

Reply:
the promise can be resolved only once, calling the second time won't do anything, you need a new deferred for each deploy